devlog/2024-02-09-Intro-to-optimizing-packages-on-solus.md

```diff
@@ -0,0 +1,488 @@
+---
+title: Intro to Optimizing Packages on Solus
+description: Explore how to employ advanced compiler techniques such as PGO, BOLT & Glibc HWCaps to squeeze extra performance from packages using libwebp as a test vehcile
+slug: solus-optimizing-packages
+authors:
+  - name: Joey Riches
+    title: Solus Staff
+    url: https://github.com/joebonrichie
+    image_url: https://avatars.githubusercontent.com/u/5338090?s=400&u=f77ed45c7e83814ce3e8bd199fc293bd5b53682b&v=4
+tags: [pgo, lto, blot, solus, packaging, optmization, 03, clang, gnu, llvm, glibc, hwcaps, x86_64-v3]
+hide_table_of_contents: false
+---
+
+We'll explore how to build packages with advanced compiler techniques in order to squeeze more performance out of the box for packages in Solus. We'll be using the story of how `libwebp` was optimized for and how it led to an unexpected side quest.
+
+<!-- truncate -->
+
+# Por Qué la Cauza
+
+Linux distributions have a lot of control over how a source-based package gets compiled and shipped to users as part of a binary repository. Aggressive and advanced compiler optimization techniques, as well as other methods can be used to provide greater out of the box performance for end users. This can greatly benefit: users running on older hardware to provide a snappier end-user experience; reducing time waiting on a heavy workload to finish; or even improved battery life; amongst other reasons.
```

Comment:
```suggestion
Linux distributions have a lot of control over how a source-based package gets compiled and shipped to users as part of a binary repository. Aggressive and advanced compiler optimization techniques, as well as other methods can be used to provide greater out of the box performance for end users. This can greatly benefit users running on older hardware to provide a snappier end-user experience; reducing time waiting on a heavy workload to finish; or even improved battery life; amongst other improvements.
```